### PR TITLE
feat: added navigation feature

### PR DESCRIPTION
**Pull request by mohsen-deriv**
- added navigation bar for mobile and desktop
- added link component
- added needed stories for hk and row navigations

### CODE REVIEW

File: libs/blocks/src/lib/navigation/desktop/desktop.blur.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import clsx from 'clsx';
+import { useEffect, useRef } from 'react';
+import { useHover } from 'usehooks-ts';
+
+export interface NavBlurProps {
+  isVisible: boolean;
```
Comment (prince-deriv) [inline libs/blocks/src/lib/navigation/desktop/desktop.blur.tsx:6]:
```suggestion
  is_visible: boolean;
```

File: libs/blocks/src/lib/navigation/desktop/desktop.blur.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  onHover?: VoidFunction;
+}
+
+const DesktopNavBlur = ({ isVisible, onHover }: NavBlurProps) => {
+  const ref = useRef<HTMLDivElement>(null);
+
+  const isHovering = useHover(ref);
```
Comment (prince-deriv) [inline libs/blocks/src/lib/navigation/desktop/desktop.blur.tsx:13]:
```suggestion
  const is_hovering = useHover(ref);
```

File: libs/blocks/src/lib/navigation/desktop/desktop.blur.tsx (hunk 1 above)
Comment (mohsen-deriv) [reply libs/blocks/src/lib/navigation/desktop/desktop.blur.tsx:6]: no more snake casing man :sweat_smile:

File: libs/blocks/src/lib/navigation/desktop/desktop.blur.tsx (hunk 2 above)
Comment (mohsen-deriv) [reply libs/blocks/src/lib/navigation/desktop/desktop.blur.tsx:13]: no more snake casing man :sweat_smile:

**Pull request merged**